src/incremental/__init__.py

```diff
@@ -11,105 +11,56 @@
 
 import sys
 import warnings
+from typing import TYPE_CHECKING, Any, TypeVar, Generic, Union, Optional, Dict
 
 #
 # Compat functions
 #
 
-if sys.version_info < (3, 0):
-    _PY3 = False
-else:
-    _PY3 = True
+_T = TypeVar("_T", contravariant=True)
 
-try:
-    _cmp = cmp
-except NameError:
 
-    def _cmp(a, b):
-        """
-        Compare two objects.
+if TYPE_CHECKING:
+    from typing_extensions import Literal
+    from distutils.dist import Distribution as _Distribution
 
-        Returns a negative number if C{a < b}, zero if they are equal, and a
-        positive number if C{a > b}.
-        """
-        if a < b:
-            return -1
-        elif a == b:
-            return 0
-        else:
-            return 1
+    def _cmp(a, b):  # type: (Any, Any) -> int
+        return 0
 
 
-def _comparable(klass):
-    """
-    Class decorator that ensures support for the special C{__cmp__} method.
```

Comment:
this decorator is broken on mypy - and the type annotations are slightly different,

eg Version is only comparable with other Versions